aws-lc-rs/src/aead.rs

```diff
@@ -396,6 +396,43 @@ impl<N: NonceSequence> SealingKey<N> {
             in_out,
         )
     }
+
+    /// Encrypts and signs (“seals”) data in place with extra plaintext.
+    ///
+    /// `aad` is the additional authenticated data (AAD), if any. This is
+    /// authenticated but not encrypted. The type `A` could be a byte slice
+    /// `&[u8]`, a byte array `[u8; N]` for some constant `N`, `Vec<u8>`, etc.
+    /// If there is no AAD then use `Aad::empty()`.
+    ///
+    /// The plaintext is given as the input value of `in_out` and `extra_in`. `seal_in_place()`
+    /// will overwrite the plaintext contained in `in_out` with the ciphertext. The `extra_in` will
+    /// be encrypted into the `extra_out_and_tag`, along with the tag.
+    /// The `extra_out_and_tag` length must be equal to the `extra_len` and `self.algorithm.tag_len()`.
```

Comment:
Technically, `extra_out_and_tag` can be longer than `extra_len + self.algorithm.tag_len()`, but only the bytes up to that length will be modified. So, the caller should ignore subsequent bytes.

Reply:
Good question... Would it be better to just match what the existing API and allow for the larger messages? Doesn't matter to me.

Reply:
I'd prefer that we enforce the length constraint as documented to avoid potential client bugs/confusion/misuse. Such a constraint could potentially be removed in the future, but if we don't enforce it now, we'll never be able to enforce it.

Reply:
Sounds good. I'll make the change